File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/entity/Event.java
package com.example.touralbum.ui.reminder.entity;

import android.os.Parcel;
import android.os.Parcelable;
import android.provider.BaseColumns;

public class Event implements BaseColumns, Parcelable{
    private Integer mId;
    private String mTitle;
    private String mContent;
    private String mCreatedTime;
    public Integer getmIsClocked() {
        return mIsClocked;
    }
    public void setmIsClocked(Integer mIsClocked) {
        this.mIsClocked = mIsClocked;
    }
    private String mUpdatedTime;
    private Integer mIsClocked = 0;
    public Event(){}
    protected Event(Parcel in) {
        if (in.readByte() == 0) {
            mId = null;
        } else {
            mId = in.readInt();
        }
        mTitle = in.readString();
        mContent = in.readString();
        mCreatedTime = in.readString();
        mUpdatedTime = in.readString();
        if (in.readByte() == 0) {
            mIsClocked = null;
        } else {
            mIsClocked = in.readInt();
        }
        if (in.readByte() == 0) {
            mIsImportant = null;
        } else {
            mIsImportant = in.readInt();
        }
        mRemindTime = in.readString();
    }

    public static final Creator<Event> CREATOR = new Creator<Event>() {
        @Override
        public Event createFromParcel(Parcel in) {
            return new Event(in);
        }

        @Override
        public Event[] newArray(int size) {
            return new Event[size];
        }
    };

    public Integer getmIsImportant() {
        return mIsImportant;
    }
    public void setmIsImportant(Integer mIsImportant) {
        this.mIsImportant = mIsImportant;
    }
    private Integer mIsImportant;
    public Integer getmId() {
        return mId;
    }
    public void setmId(Integer mId) {
        this.mId = mId;
    }
    public String getmTitle() {
        return mTitle;
    }
    public void setmTitle(String mTitle) {
        this.mTitle = mTitle;
    }
    public String getmContent() {
        return mContent;
    }
    public void setmContent(String mContent) {
        this.mContent = mContent;
    }
    public String getmCreatedTime() {
        return mCreatedTime;
    }
    public void setmCreatedTime(String mCreatedTime) {
        this.mCreatedTime = mCreatedTime;
    }
    public String getmUpdatedTime() {
        return mUpdatedTime;
    }
    public void setmUpdatedTime(String mUpdatedTime) {
        this.mUpdatedTime = mUpdatedTime;
    }
    public String getmRemindTime() {
        return mRemindTime;
    }
    public void setmRemindTime(String mRemindTime) {
        this.mRemindTime = mRemindTime;
    }
    private String mRemindTime;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        if (mId == null) {
            dest.writeByte((byte) 0);
        } else {
            dest.writeByte((byte) 1);
            dest.writeInt(mId);
        }
        dest.writeString(mTitle);
        dest.writeString(mContent);
        dest.writeString(mCreatedTime);
        dest.writeString(mUpdatedTime);
        if (mIsClocked == null) {
            dest.writeByte((byte) 0);
        } else {
            dest.writeByte((byte) 1);
            dest.writeInt(mIsClocked);
        }
        if (mIsImportant == null) {
            dest.writeByte((byte) 0);
        } else {
            dest.writeByte((byte) 1);
            dest.writeInt(mIsImportant);
        }
        dest.writeString(mRemindTime);
    }
}

File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/entity/Event.kt
package com.example.touralbum.ui.reminder.entity

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import android.provider.BaseColumns

class Event : BaseColumns, Parcelable {
    private var mId: Int? = null
    private var mTitle: String? = null
    private var mContent: String? = null
    private var mCreatedTime: String? = null
    fun getmIsClocked(): Int? {
        return mIsClocked
    }

    fun setmIsClocked(mIsClocked: Int?) {
        this.mIsClocked = mIsClocked
    }

    private var mUpdatedTime: String? = null
    private var mIsClocked: Int? = 0

    constructor() {}
    protected constructor(`in`: Parcel) {
        mId = if (`in`.readByte().toInt() == 0) {
            null
        } else {
            `in`.readInt()
        }
        mTitle = `in`.readString()
        mContent = `in`.readString()
        mCreatedTime = `in`.readString()
        mUpdatedTime = `in`.readString()
        mIsClocked = if (`in`.readByte().toInt() == 0) {
            null
        } else {
            `in`.readInt()
        }
        mIsImportant = if (`in`.readByte().toInt() == 0) {
            null
        } else {
            `in`.readInt()
        }
        mRemindTime = `in`.readString()
    }

    fun getmIsImportant(): Int? {
        return mIsImportant
    }

    fun setmIsImportant(mIsImportant: Int?) {
        this.mIsImportant = mIsImportant
    }

    private var mIsImportant: Int? = null
    fun getmId(): Int? {
        return mId
    }

    fun setmId(mId: Int?) {
        this.mId = mId
    }

    fun getmTitle(): String? {
        return mTitle
    }

    fun setmTitle(mTitle: String?) {
        this.mTitle = mTitle
    }

    fun getmContent(): String? {
        return mContent
    }

    fun setmContent(mContent: String?) {
        this.mContent = mContent
    }

    fun getmCreatedTime(): String? {
        return mCreatedTime
    }

    fun setmCreatedTime(mCreatedTime: String?) {
        this.mCreatedTime = mCreatedTime
    }

    fun getmUpdatedTime(): String? {
        return mUpdatedTime
    }

    fun setmUpdatedTime(mUpdatedTime: String?) {
        this.mUpdatedTime = mUpdatedTime
    }

    fun getmRemindTime(): String? {
        return mRemindTime
    }

    fun setmRemindTime(mRemindTime: String?) {
        this.mRemindTime = mRemindTime
    }

    private var mRemindTime: String? = null
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        if (mId == null) {
            dest.writeByte(0.toByte())
        } else {
            dest.writeByte(1.toByte())
            dest.writeInt(mId!!)
        }
        dest.writeString(mTitle)
        dest.writeString(mContent)
        dest.writeString(mCreatedTime)
        dest.writeString(mUpdatedTime)
        if (mIsClocked == null) {
            dest.writeByte(0.toByte())
        } else {
            dest.writeByte(1.toByte())
            dest.writeInt(mIsClocked!!)
        }
        if (mIsImportant == null) {
            dest.writeByte(0.toByte())
        } else {
            dest.writeByte(1.toByte())
            dest.writeInt(mIsImportant!!)
        }
        dest.writeString(mRemindTime)
    }

    companion object {
        @JvmField val CREATOR: Creator<Event?> = object : Creator<Event?> {
            override fun createFromParcel(`in`: Parcel): Event? {
                return Event(`in`)
            }

            override fun newArray(size: Int): Array<Event?> {
                return arrayOfNulls(size)
            }
        }
    }
}